packages/viewer/src/scenes/camera.ts

```diff
@@ -158,7 +160,9 @@ export class Camera implements FrameCamera.FrameCamera {
    * Queues the rendering for a new frame using this camera. The returned
    * promise will resolve when a frame is received that contains this camera.
    */
-  public async render(renderOptions?: CameraRenderOptions): Promise<void> {
+  public async render(
+    renderOptions?: CameraRenderOptions
+  ): Promise<StreamEvents | undefined> {
```

Comment:
I think we need to think about these events a bit differently and perhaps a bit more abstractly. It's possible that we add additional events in the future, and as a developer, I'd want to subscribe to certain ones. We actually have two types of events right now, one for the first frame being received and this new one for animation end. What if we had an API and convention like this:

```ts
const camera = (await viewer.scene()).camera();
const resp = await camera
  .update({ position: {x: 1, y: 1, z: 1 })
  .render({ animation: {milliseconds: 1000} }); // Completes when FSS sends a response.

// Any data that we want to expose from the FSS response.
console.log(resp.data);

// Invokes the callback once and removes the listener. Need to think about 
// how event listeners are easily cleaned up to prevent a memory leak.
resp.onAnimationCompleted.once(() => console.log('animation done'));

// Invokes the callback N times for each animationCompleted event.
const handler = () => console.log('animation done')
resp.onAnimationCompleted.on(handler);

// Remove the event listener.
resp.onAnimationCompleted.off(handler);
```

BTW, there's a shell for a `EventDispatcher` class that is we may be able to leverage.